### ISSUE TITLE
 imported but not used (typecheck)

### ISSUE THREAD

**Issue by zhaoyunxing92** [golangci/golangci-lint#2862]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

typecheck linter reports false positive errors on imports for packages which have name different from the module name.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
1.41.1
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
# paste output here
linters-settings:
  govet:
    check-shadowing: true
  golint:
    min-confidence: 0
  gocyclo:
    min-complexity: 10
  maligned:
    suggest-new: true
  dupl:
    threshold: 100
  goconst:
    min-len: 2
    min-occurrences: 2
  depguard:
    list-type: blacklist
    packages:
      # logging is allowed only by logutils.Log, logrus
      # is allowed to use only in logutils package
      - github.com/sirupsen/logrus
  misspell:
    locale: US
  lll:
    line-length: 140
  goimports:
    local-prefixes: github.com/golangci/golangci-lint
  gocritic:
    enabled-tags:
      - performance
      - style
      - experimental
    disabled-checks:
      - wrapperFunc

linters:
  disable-all: true
  enable:
    - govet
    - staticcheck
    - ineffassign
    - misspell

run:
  skip-dirs:
    - test/testdata_etc
    - pkg/golinters/goanalysis/(checker|passes)

issues:
  exclude-rules:
    - text: "weak cryptographic primitive"
      linters:
        - gosec
    - linters:
        - staticcheck
      text: "SA1019:"

# golangci.com configuration
# https://github.com/golangci/golangci/wiki/Configuration
service:
  golangci-lint-version: 1.15.x # use the fixed version to not introduce new linters unexpectedly
  prepare:
    - echo "here I can run custom commands, but no preparation needed for this repo"
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
1.13

GO111MODULE=""
  GOARCH="amd64"
  GOBIN=""
  GOCACHE="/home/runner/.cache/go-build"
  GOENV="/home/runner/.config/go/env"
  GOEXE=""
  GOFLAGS=""
  GOHOSTARCH="amd64"
  GOHOSTOS="linux"
  GONOPROXY=""
  GONOSUMDB=""
  GOOS="linux"
  GOPATH="/home/runner/go"
  GOPRIVATE=""
  GOPROXY="[https://proxy.golang.org,direct](https://proxy.golang.org%2Cdirect/)"
  GOROOT="/opt/hostedtoolcache/go/1.13.15/x64"
  GOSUMDB="sum.golang.org"
  GOTMPDIR=""
  GOTOOLDIR="/opt/hostedtoolcache/go/1.13.15/x64/pkg/tool/linux_amd64"
  GCCGO="gccgo"
  AR="ar"
  CC="gcc"
  CXX="g++"
  CGO_ENABLED="1"
  GOMOD=""
  CGO_CFLAGS="-g -O2"
  CGO_CPPFLAGS=""
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here

Run golangci-lint run --timeout=10m -v
level=info msg="[config_reader] Config search paths: [./ /home/runner/work/dubbo-go/dubbo-go /home/runner/work/dubbo-go /home/runner/work /home/runner /home /]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 4 linters: [govet ineffassign misspell staticcheck]"
level=info msg="[loader] Go packages loading at mode 575 (deps|exports_file|name|types_sizes|compiled_files|files|imports) took 2m10.555005976s"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 56.346[8](https://github.com/apache/dubbo-go/runs/6475077184?check_suite_focus=true#step:8:8)18ms"
level=info msg="[linters context/goanalysis] analyzers took 1m50.437648188s with top 10 stages: buildir: 1m31.46[9](https://github.com/apache/dubbo-go/runs/6475077184?check_suite_focus=true#step:8:9)77433s, nilness: 7.131271175s, inspect: 2.552770752s, fact_deprecated: 1.464938398s, ctrlflow: 1.464382999s, printf: 1.128907497s, fact_purity: 987.689019ms, typedness: 947.686229ms, SA5012: 634.748186ms, misspell: 317.228465ms"
level=info msg="[runner/skip dirs] Skipped 15 issues from dir protocol/grpc/protoc-gen-dubbo/examples by pattern (^|/)examples($|/)"
Error: remoting/etcdv3/listener.go:[10](https://github.com/apache/dubbo-go/runs/6475077184?check_suite_focus=true#step:8:10)1:45: undeclared name: `clientv3` (typecheck)
level=info msg="[runner] Issues before processing: 238, after processing: 2"
level=info msg="[runner] Processors filtering stat (out/in): max_from_linter: 2/2, sort_results: 2/2, autogenerated_exclude: 208/223, exclude-rules: 199/208, max_per_file_from_linter: 2/2, path_prefixer: 2/2, skip_dirs: 223/238, nolint: 198/199, uniq_by_line: 2/198, diff: 2/2, source_code: 2/2, path_shortener: 2/2, path_prettifier: 238/238, skip_files: 238/238, identifier_marker: 208/208, exclude: 208/208, max_same_issues: 2/2, severity-rules: 2/2, cgo: 238/238, filename_unadjuster: 238/238"
level=info msg="[runner] processing took 15.32565ms with stages: exclude-rules: 7.847936ms, identifier_marker: 4.545552ms, nolint: 1.8004ms, path_prettifier: 413.823µs, autogenerated_exclude: 313.817µs, skip_dirs: 227.8[12](https://github.com/apache/dubbo-go/runs/6475077184?check_suite_focus=true#step:8:12)µs, filename_unadjuster: 54.403µs, source_code: 47.303µs, cgo: 46.003µs, uniq_by_line: 15.901µs, max_same_issues: 4.3µs, max_per_file_from_linter: 2.2µs, path_shortener: 2.1µs, max_from_linter: 1.9µs, exclude: 600ns, diff: 500ns, skip_files: 400ns, severity-rules: 300ns, sort_results: 300ns, path_prefixer: 100ns"
level=info msg="[runner] linters took 1m5.429969576s with stages: goanalysis_metalinter: 1m5.4[14](https://github.com/apache/dubbo-go/runs/6475077184?check_suite_focus=true#step:8:14)485817s"
level=info msg="File cache stats: 506 entries of total size 2.0MiB"
level=info msg="Memory: 1962 samples, avg is 396.8MB, max is 1491.6MB"
func (l *EventListener) handleEvents(event *clientv3.Event, listeners ...remoting.DataListener) bool {
                                            ^
level=info msg="Execution took 3m16.059640818s"
Error: remoting/etcdv3/listener.go:31:2: "go.etcd.io/etcd/client/v3" imported but not used (typecheck)
	"go.etcd.io/etcd/client/v3"
	^
Error: Process completed with exit code 1.
```

</details>

### Code example or link to a public repository

<details>

```go
// add your code here

https://github.com/apache/dubbo-go/blob/6960942bf45c8d5a5e5e7fa5a46edf26aef6337e/remoting/etcdv3/listener.go#L31

https://github.com/apache/dubbo-go/blob/6960942bf45c8d5a5e5e7fa5a46edf26aef6337e/remoting/etcdv3/listener.go#L101
```
[#L31](https://github.com/apache/dubbo-go/blob/6960942bf45c8d5a5e5e7fa5a46edf26aef6337e/remoting/etcdv3/listener.go#L31
)

[#L101](https://github.com/apache/dubbo-go/blob/6960942bf45c8d5a5e5e7fa5a46edf26aef6337e/remoting/etcdv3/listener.go#L101)
</details>
